dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ItemAccessStatusLinkRepository.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+
+package org.dspace.app.rest.repository;
+
+import java.sql.SQLException;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import javax.servlet.http.HttpServletRequest;
+
+import org.dspace.access.status.service.AccessStatusService;
+import org.dspace.app.rest.model.AccessStatusRest;
+import org.dspace.app.rest.model.ItemRest;
+import org.dspace.app.rest.projection.Projection;
+import org.dspace.content.Item;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.stereotype.Component;
+
+/**
+ * Link repository for calculating the access status of an Item
+ */
+@Component(ItemRest.CATEGORY + "." + ItemRest.NAME + "." + ItemRest.ACCESS_STATUS)
+public class ItemAccessStatusLinkRepository extends AbstractDSpaceRestRepository
+    implements LinkRestRepository {
+
+    @Autowired
+    ItemService itemService;
+
+    @Autowired
+    AccessStatusService accessStatusService;
+
+    public AccessStatusRest getAccessStatus(@Nullable HttpServletRequest request,
```

Comment:
I removed the `@PreAuthorize` annotation as it was causing some weird behaviors (errors in the browser console) on the UI side for workflow items. I tryed to manage those errors (401 even if I was connected as an administrator), but could not find a way to do it. The status is also public and can be displayed to any user so I thought it was fine to remove the restriction for this specific case.

Reply:
@nibou230 : The status here should not actually be "public" at all times.  If an embargo is placed at the Item-level or an Item is Admin-only, then then entire item should be "invisible" to anonymous users.  If the `accessStatus` endpoint still responds, then it would verify the existence of the Item.  While this isn't necessarily a major security issue, it's unexpected behavior -- as anonymous users should not be able to verify the existence of admin-only Items (and similar).

We might want to look closer at why the `@PreAuthorize` isn't working in this scenario. It should work for any Item*LinkRepository object, as we have it working for these objects:
* https://github.com/DSpace/DSpace/blob/main/dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ItemBundleLinkRepository.java
* https://github.com/DSpace/DSpace/blob/main/dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ItemRelationshipLinkRepository.java
* https://github.com/DSpace/DSpace/blob/main/dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ItemThumbnailLinkRepository.java

So, if it's not working here, that'd imply there may be a flaw in this code (or the UI code) which we are not seeing.  In summary, I think we need to find a way to enable this `@PreAuthorize` as otherwise it won't behave like other similar endpoints.

Reply:
That's OK, I will bring back the `@PreAuthorize`, but I still don't know why I get some weird 401 in the page that contains the workflow items. It works fine, as the status is properly displayed, but I don't think we want to leave this behavior in place. Can someone help me look at this problem and adapt the code if needed? I will do my best, but I don't know if I will have the time to investiguate this after tomorrow.

Reply:
@nibou230 : I might be able to help at some point, but I cannot give a timeline right now... currently I'm swamped with other reviews for 7.3.  But maybe someone else will chime in here with support once this is updated.

Reply:
@nibou230 : I retested this today with the `@PreAuthorize` in place. I'm not able to reproduce any 401 errors on the MyDSpace page when accessing workspace/workflow items.  So, I believe that issue you were seeing previously is either fixed, or it wasn't caused by the `@PreAuthorize()`